### PR TITLE
Better error message for canvas submission failures

### PR DESCRIPTION
**Pull request by marcospri**
### TODO

- [ ] Fix tests
- [ ] Allow to see the content in the back (ie not hide the background with the loaded assigment)

### Testing

if you want to repro the error code, the easiest path is to apply a diff like

```diff
diff --git a/lms/views/api/grading.py b/lms/views/api/grading.py
index a8a7dbceb..26d894450 100644
--- a/lms/views/api/grading.py
+++ b/lms/views/api/grading.py
@@ -96,6 +96,9 @@ class GradingViews:
 
         [1] https://canvas.instructure.com/doc/api/file.assignment_tools.html
         """
+        raise SerializableError(
+            error_code=ErrorCode.CANVAS_SUBMISSION_COURSE_NOT_AVAILABLE
+        )
 
         # Send the SpeedGrader LTI launch URL to the Canvas instance, if we haven't
         # already created a submission OR if the existing submission has not been
```

- Reload [an assignment](https://hypothesis.instructure.com/courses/319/assignments/4701) **as a student** and make one annotation.

### CODE REVIEW

File: lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx [hunk 1]
```diff
@@ -327,6 +327,11 @@ export default function BasicLTILaunchApp() {
           errorState={errorState}
           error={error}
           onRetry={authorizeAndFetchURL}
+          /* istanbul ignore next */
```
Comment (marcospri) [inline lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.tsx:330]: IGNORE IT!!! (it doesn't)

File: lms/static/scripts/frontend_apps/components/LaunchErrorDialog.tsx [hunk 2]
```diff
@@ -611,17 +615,51 @@ export default function LaunchErrorDialog({
...
+            {' '}
+            <b>
+              Your instructor is not aware you have saved an annotation;
+            </b>{' '}
+            you might need to reach out to them.
+          </p>
+        </ErrorModal>
+      );
+
+    case 'canvas_submission_course_not_available':
```
Comment (marcospri) [inline lms/static/scripts/frontend_apps/components/LaunchErrorDialog.tsx:642]: New error message, adds a hint about the likely cause.

File: lms/static/scripts/frontend_apps/components/LaunchErrorDialog.tsx [hunk 3]
```diff
@@ -611,17 +615,51 @@ export default function LaunchErrorDialog({
...
+          <p>
+            {' '}
+            Hypothesis saved your annotation, but was unable to generate a
+            submission.
+          </p>
+
+          <p>
+            {' '}
+            <b>
+              Your instructor is not aware you have saved an annotation;
```
Comment (marcospri) [inline lms/static/scripts/frontend_apps/components/LaunchErrorDialog.tsx:635]: Existing dialog, add more details about what just happend.

File: lms/static/scripts/frontend_apps/components/LaunchErrorDialog.tsx [hunk 4]
```diff
@@ -611,17 +615,51 @@ export default function LaunchErrorDialog({
...
 
     case 'error-reporting-submission':
-      // nb. There is no retry action here as we just suggest reloading the entire
-      // page.
       return (
         <ErrorModal
           {...defaultProps}
           onRetry={undefined}
-          title="Something went wrong"
+          onCancel={onCancel}
```
Comment (marcospri) [inline lms/static/scripts/frontend_apps/components/LaunchErrorDialog.tsx:622]: Wording from:

https://hypothes-is.slack.com/archives/C2C2U40LW/p1716385676586839?thread_ts=1716217003.396489&cid=C2C2U40LW


https://hypothes-is.slack.com/archives/C2C2U40LW/p1716385676586839?thread_ts=1716217003.396489&cid=C2C2U40LW


Although I'd vote to merge any version  with the code changes to enable the button & background and then do a another PR if the text needs to be revisited.

**Comment by robertknight** [comment]
Superseded by https://github.com/hypothesis/lms/pull/6384.
